conans/client/build/meson.py

```diff
@@ -164,7 +164,11 @@ def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
             build_type
         ])
         command = 'meson "%s" "%s" %s' % (source_dir, self.build_dir, arg_list)
-        with environment_append({"PKG_CONFIG_PATH": pc_paths}):
+        envs = {"PKG_CONFIG_PATH": pc_paths}
+        if self._append_vcvars and str.startswith(self.backend, "vs"):
+            vcvars_dict = tools.vcvars_dict(self._settings, output=self._conanfile.output)
```

Comment:
This change might be breaking. It introduces the usage of vcvars in an scenario that was not done before.

Reply:
This change is because when meson uses vs as the backend to generate vs solution, it is in the generation configuration stage. It happens that if vcvars is not attached, meson will find vcvars by itself, and then the 32-bit arch set will be invalid. In ninja, Environment variables are used in the generation phase, so ninja inserts vcvars when building.

E.g:
Set compiler.arch=x86
Then using vs as the backend, the 64-bit program is generated instead of the expected 32-bit program.
If you add this patch, VS will generate a 32-bit program.